### PR TITLE
*: improve selfhosted test

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Test failure might be caused by left over pods from other tests.
Basically each pod will occupy one host.
I just clean up this morning and will rerun it.

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

File: test/e2e/self_hosted_test.go [hunk 1]
```diff
@@ -74,7 +75,8 @@ func TestCreateSelfHostedClusterWithBootMember(t *testing.T) {
 	defer e.Close()
 
 	<-e.Server.ReadyNotify()
-	fmt.Println("etcdserver is ready")
+
+	t.Log("etcdserver is ready")
```
Comment (hongchaodeng) [inline test/e2e/self_hosted_test.go]: t.Log() will only show after the test finished.
This is not really helpful: 1. debugging; 2. mixing logs with others that already use more real-time logging like fmt, logrus.

Can we keep fmt? I don't mind to change to logrus.

File: test/e2e/self_hosted_test.go [hunk 2]
```diff
@@ -53,14 +53,15 @@ func TestCreateSelfHostedClusterWithBootMember(t *testing.T) {
 	os.RemoveAll(dir)
 	defer os.RemoveAll(dir)
 
+	host, _ := netutil.GetDefaultHost()
+
```
Comment (hongchaodeng) [inline test/e2e/self_hosted_test.go]: For debugging sake, can we print this and also change "hack/test" to only run this test?

File: test/e2e/self_hosted_test.go (hunk 1 above)
Comment (xiang90) [inline test/e2e/self_hosted_test.go]: print in test is wrong. test should generate no output if it succeeds. if you want to debug, add debugging line and disable by default.

When test fails, it should print out log. So using https://golang.org/pkg/testing/#T.Log is correct. And this is also why t.log designed this way at the first place.

File: test/e2e/self_hosted_test.go (hunk 2 above)
Comment (xiang90) [inline test/e2e/self_hosted_test.go]: we should print out nothing in test. i can add t.log if you wish to.

Comment (hongchaodeng) [inline test/e2e/self_hosted_test.go]: Let's not add t.Log here.
Could you check err?

File: test/e2e/self_hosted_test.go (hunk 1 above)
Comment (hongchaodeng) [inline test/e2e/self_hosted_test.go]: OK

**Comment by hongchaodeng** [comment]
I think the error might be caused by that self hosted pods in another test not "removed" yet due to async nature.
There has one pod created. We have exactly 4 worker nodes.
I need to further debug this.

**Comment by hongchaodeng** [comment]
I need to rewrite e2e tests as mentioned in https://github.com/coreos/etcd-operator/issues/385#issuecomment-261753751 to pass the tests again.

The failure here isn't relevant and it would be good to have such refactor work in. Maybe we could just merge this? Do you have any opinion?

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

Also fixed the firewall issue of backup pod port "19999".

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

I shut down proxy before. Now restarted. Should work.

**Comment by hongchaodeng** [comment]
tests passed.
Merging.

**Pull request merged**